### ISSUE TITLE
REQUEST: Create kubernetes-sigs/noderesourcetopology-api

### ISSUE THREAD

**Issue by swatisehgal** [kubernetes/org#4224]
### New repo, staging repo, or migrate existing

new repository

### Is it a staging repo?

no

### Requested name for new repository

noderesourcetopology-api

### Which Organization should it reside

kubernetes-sigs

### Who should have admin access?

swatisehgal, ffromani

### Who should have write access?

marquiz

### Who should be listed as approvers in OWNERS?

swatisehgal, ffromani, marquiz

### Who should be listed in SECURITY_CONTACTS?

swatisehgal, ffromani, marquiz

### What should the repo description be?

CRD based API to capture Node Resource Topology information

### What SIG and subproject does this fall under?

sig-node, subproject: resource-management

### Please provide references to appropriate approval for this new repo

1. Input received during API review: https://github.com/kubernetes/kubernetes/pull/96275#discussion_r1119165747. Previously the API was being proposed under staging but that proposal was rejected and it was recommended to propose this under kubernetes-sigs.
2. Had a discussion in SIG Node meeting on [23/05/2023](https://docs.google.com/document/d/1Ne57gvidMEWXR70OxxnRkYquAoMpt56o75oZtg-OeBg/edit#bookmark=id.a0pn2fo524i1) to get everyone upto speed on the input received from the API reviewers and identify the best course of action for this repo. 

### Additional context for request

1. [NFD Topology Updater](https://github.com/kubernetes-sigs/node-feature-discovery/blob/master/docs/get-started/introduction.md#nfd-topology-updater) in [kubernetes-sigs/node-feature-discovery](https://github.com/kubernetes-sigs/node-feature-discovery) and [Topology aware scheduler plugin](https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/pkg/noderesourcetopology) in [kubernetes-sigs/scheduler-plugins](https://github.com/kubernetes-sigs/scheduler-plugins) both depend on this API.
2. Issue: https://github.com/kubernetes/community/issues/6308
3. Related work: https://github.com/kubernetes/kubernetes/pull/110252 and https://github.com/kubernetes/kubernetes/pull/96275

**Comment by ffromani** [comment]
thanks @swatisehgal . Fully makes sense to me because both https://github.com/kubernetes-sigs/scheduler-plugins and https://github.com/kubernetes-sigs/node-feature-discovery both depend on this API.

**Comment by pacoxu** [comment]
Should this include sig scheduling?

**Comment by swatisehgal** [comment]
> Should this include sig scheduling?

@pacoxu I believe that this falls primarily under SIG Node as this API exposes node specific information that can be used for various use cases one of which is Topology aware scheduling. There could be other components that could use this API such as DRA controller.

Having said I would be more than happy to have input from sig scheduling on this.

**Comment by kad** [comment]
+1 for creating that repository and moving development of those CRDs under official Kubernetes umbrella.

**Comment by swatisehgal** [comment]
/sig node
(as the primary SIG)

/sig scheduling 
(for input)

**Comment by liggitt** [comment]
A sigs repo for this makes sense to me, so 👍 

Since this contains a k8s.io API which [falls under API review](https://go.k8s.io/api-review#what-apis-need-to-be-reviewed), please ensure the directories containing the API definitions (go types and/or CRD manifests) require [API approval](https://github.com/kubernetes/kubernetes/blob/master/OWNERS_ALIASES#LL318C3-L318C17) to merge changes to those definitions

**Comment by mrbobbytables** [comment]
/assign @dchen1107 @derekwaynecarr @mrunalp 
for signoff from sig node TLs

**Comment by SergeyKanzhelev** [comment]
> /assign @dchen1107 @derekwaynecarr @mrunalp for signoff from sig node TLs

It was discussed at [SIG Node meeting at 05/23](https://docs.google.com/document/d/1Ne57gvidMEWXR70OxxnRkYquAoMpt56o75oZtg-OeBg/edit#bookmark=id.a0pn2fo524i1) and approved. @dchen1107 @derekwaynecarr @mrunalp can one of you formally approve it here?

**Comment by dchen1107** [comment]
/approve 

Discussed and reviewed and approved at SIG Node meeting on 5/23. Thanks!

**Comment by mrbobbytables** [comment]
I'm confused - the repo does exist already and has been populated. What is the action being requested? :x

**Comment by swatisehgal** [comment]
> I'm confused - the repo does exist already and has been populated. 

Previously the API was being proposed under staging and was [bootstrapped](https://github.com/kubernetes/kubernetes/pull/110252) with owner file/ readme etc. The API was NOT populated initially. During the review process of the follow-up PR that tried to populate the API, the proposal was [rejected](https://github.com/kubernetes/kubernetes/pull/96275#discussion_r1119165747) and it was recommended to propose the API under kubernetes-sigs and we are requesting a repo to be created (under kubernetes-sigs) in this issue for that.

I pivoted that [PR](https://github.com/kubernetes/kubernetes/pull/96275) to remove the repo from staging based on the input received. I am not sure why https://github.com/kubernetes/noderesourcetopology-api/ wasn't removed in spite of its removal from k/k/staging. 

Hope this helps.

> What is the action being requested? :x

This request is to create a repo under kubernetes-sigs which can be populated with the CRD based API in the coming release(s) with reviews from other community members and API reviewers.

**Comment by mrbobbytables** [comment]
> I pivoted that [PR](https://github.com/kubernetes/kubernetes/pull/96275) to remove the repo from staging based on the input received. I am not sure why https://github.com/kubernetes/noderesourcetopology-api/ wasn't removed in spite of its removal from k/k/staging.

staging repos are not removed automatically, they have to be cleaned up separately

So - just to be clear, the repo is 100% good to delete and recreate, correct? - I just don't want to inadvertently delete something that is needed

**Comment by swatisehgal** [comment]
Yes, please go ahead and delete https://github.com/kubernetes/noderesourcetopology-api/.

The new repo should be created under kubernetes-sigs: https://github.com/kubernetes-sigs/noderesourcetopology-api.

**Comment by mrbobbytables** [comment]
update test-infra to remove staging rule for previous repo: https://github.com/kubernetes/test-infra/pull/30037
repo created: https://github.com/kubernetes-sigs/noderesourcetopology-api
teams added: https://github.com/kubernetes/org/pull/4323
k/community update: https://github.com/kubernetes/community/pull/7390

Once those PRs are merged I'll add the teams to the repo and finish removing the staging repo.

**Comment by mrbobbytables** [comment]
teams have been added - just waiting on the branch protection rule to be removed before I archive the staging repo
